lib/core/include/irods/irods_server_properties.hpp

```diff
@@ -170,10 +170,21 @@ namespace irods
         };
 
     public:
-        /// \brief Acquire a config_handle object containing a const reference to the underlying json object.
-        config_handle map() const noexcept
+        /// \brief Acquire a readonly_config_handle object containing a const reference to the underlying json object.
+        ///
+        /// This function is used to acquire a read-only handler, to then access the underlying nlohmann::json object
+        /// containing the server's properties.
+        ///
```

Comment:
Better, but it still needs a few words explaining why the handle is necessary. For example, you could say something like:
```
The readonly_config_handle guarantees that, for the lifetime of the handle, no other
thread can modify the object returned by config_handle::get_json().
```